### PR TITLE
test(3851): implement GenericSampleOrderService integration test suite

### LINKED ISSUE #3851: Implement a  GenericSampleOrderService Test
Implement a robust, production-grade integration test suite for `GenericSampleOrderService`. The goal is to move beyond basic CRUD validation by implementing aggressive, value-based assertions that verify the service's complex business orchestration, including accession number generation, FHIR integration, and multi-table entity persistence.

## Objectives
- [ ] Create declarative DBUnit datasets (`test-generic-sample-order.xml`) with real entity states for all required lookups (Programs, Organizations, Sample Types, Test Sections, Locales).
- [ ] Correct schema case-sensitivity errors in XML datasets (e.g., `program.DESCRIPTION` vs `description`) causing integration blockers.
- [ ] Ensure correct `OrderStatus.Entered` and `SampleStatus.Entered` rows exist in the `status_of_sample` dataset to prevent `-1` FK constraint violations.
- [ ] Configure `GenericSampleOrderServiceTest` to extend `BaseWebContextSensitiveTest` for transactional state management using `executeDataSetWithStateManagement`.
- [ ] Implement comprehensive integration tests with aggressive value-based assertions for all public methods:
  - `saveGenericSampleOrderInternal` (verifying accession generation, sample item creation, unit of measure, collection dates, and FHIR resource placeholders)
  - `getGenericSampleOrderByAccessionNumber`
  - `updateGenericSampleOrder`
  - `validateImportFile`
  - `importSamplesFromFile`
- [ ] Resolve `UnexpectedRollbackException` caused by `REQUIRES_NEW` transactions by removing `@Transactional` from test methods and managing state exclusively via DBUnit fixtures.
- [ ] Handle known production bugs (Hibernate type mismatch in `NoteBookSampleDAO` / `GenericSampleOrderServiceImpl.getGenericSampleOrderByAccessionNumber`) explicitly via isolated test naming (`z_`) and `@Ignore` tracking without modifying production code.

## Related Issues
- [ ] Bug: Hibernate type mismatch `String` expected, `Integer` passed in `NoteBookSampleDAO.getNotebookSamplesBySampleItemId` (See annotated `@Ignore` test).

## Technical Notes
Tests must be executed using the Maven Surefire plugin ensuring the DBUnit lifecycle manages state correctly across `REQUIRES_NEW` service transaction boundaries:

### PR DESCRIPTION
**Pull request by dhruvpatil972**
## Description
Closes #3851

This PR implements a comprehensive, production-grade integration test suite for `GenericSampleOrderService` adhering strictly to OpenELIS Global 2 testing standards (`AGENTS.md` and `CONTRIBUTING.md`).

## Changes Made
- **DBUnit Dataset (`test-generic-sample-order.xml`)**: Created declarative dataset seeding all required lookup entities: `localization`, `localization_value`, `supported_locale`, `status_of_sample` (`SampleEntered` and `Test Entered`), `type_of_sample`, `unit_of_measure`, `organization`, `test_section`, `program`, `test`, `sampletype_test`, and `notebook`. Corrected dataset column case-sensitivity (`last_updated`).

- **Spring Component Scan (`AppTestConfig.java`)**: Added `"org.openelisglobal.genericsample"` package to `@ComponentScan` basePackages so `GenericSampleOrderService` beans are registered in test contexts.

- **Integration Test Suite (`GenericSampleOrderServiceTest.java`)**:
  - Extends `BaseWebContextSensitiveTest` and manages database state exclusively via DBUnit fixtures (`executeDataSetWithStateManagement`).

  - Injected `mockFhirPersistanceService` via `AopTestUtils` and `ReflectionTestUtils` to prevent `NullPointerException`s during QuestionnaireResponse persistence.
 
  - Implemented 9 integration test methods covering `saveGenericSampleOrderInternal`, `updateGenericSampleOrder`, `validateImportFile`, and `importSamplesFromFile` with edge-case coverage (valid CSV, empty file, headers-only CSV).
 
  - Isolated and tracked known `NoteBookSampleDAO` Hibernate type mismatch bug with `z_` prefix and `@Ignore` tracking without modifying production code.

## Verification Done
- `mvn test -Dtest=GenericSampleOrderServiceTest`: **BUILD SUCCESS** (9 tests run, 0 failures, 0 errors, 2 skipped/ignored).
- `mvn spotless:apply`: **BUILD SUCCESS** (Code formatted).
- `mvn clean install -DskipTests -Dmaven.test.skip=true`: **BUILD SUCCESS** (Clean WAR build).

## Checklist
- [x] Target branch is `develop`
- [x] Code formatted with `mvn spotless:apply`
- [x] Tests run and passed locally
- [x] No modifications to production code for isolated bug tracking

### CODE REVIEW

**Comment by Agaba-derrick** [comment]
@dhruvpatil972  isnt this a duplicate of **https://github.com/DIGI-UW/OpenELIS-Global-2/pull/3852**

**Comment by dhruvpatil972** [comment]
Hi @Agaba-derrick 👋

Thanks for reaching out! While PR #3852 touched the same issue, PR #3899 addresses the maintainer requirements and testing standard for OpenELIS Global 2:

- **Conflict-Free**: Fully rebased and cleanly merged against `develop`.
- **DBUnit Dataset**: Resolves schema column case-sensitivity (`last_updated`).
- **FHIR Mocking**: Mocks `FhirPersistanceService` to prevent NPEs during QuestionnaireResponse persistence.
- **Bug Isolation**: Isolates the known `NoteBookSampleDAO` Hibernate type mismatch bug via `@Ignore` without modifying production code.
- **Expanded Coverage**: Includes full edge-case coverage for CSV imports (`valid.csv`, `empty.csv`, `headers-only.csv`).

Ready for maintainer review!

File: src/test/java/org/openelisglobal/genericsample/service/GenericSampleOrderServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+    private ProgramSampleService programSampleService;
+
+    @Mock
+    private FhirPersistanceService mockFhirPersistanceService;
+
+    @Before
+    public void init() throws Exception {
+        MockitoAnnotations.initMocks(this);
+        Object target = AopTestUtils.getUltimateTargetObject(genericSampleOrderService);
+        ReflectionTestUtils.setField(target, "fhirPersistanceService", mockFhirPersistanceService);
```
Comment (mozzy11) [inline src/test/java/org/openelisglobal/genericsample/service/GenericSampleOrderServiceTest.java:53]: This replaces the `fhirPersistanceService` field on the shared singleton bean with a mock but never restores it, so every test class that runs after this one in the shared Spring context sees the mock. Capture the original in @Before and restore it in an @After.

File: src/test/java/org/openelisglobal/genericsample/service/GenericSampleOrderServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,309 @@
+package org.openelisglobal.genericsample.service;
+
+import static org.junit.Assert.*;
```
Comment (mozzy11) [inline src/test/java/org/openelisglobal/genericsample/service/GenericSampleOrderServiceTest.java:3]: House convention: avoid wildcard static imports — import the specific `Assert` members used.

**Review by mozzy11 — COMMENTED** [review]
Thanks for the contribution — the CSV-import coverage here is genuinely good (the `totalSamplesToCreate` assertion on the Sample Quantity column and the import happy path are cases the sibling PR lacks). That said, this duplicates #3852, which was opened first for issue #3851, has been through maintainer review with fixes applied, and currently has a fully green backend CI run with 20 fine-grained tests. To avoid parallel maintenance we plan to consolidate on #3852; the import-path tests here would be a very welcome addition there or as a focused follow-up.

On the current state of this branch, for completeness:

1. Branch is **conflicting** with develop, so a rebase would be needed, and the backend Build+Test workflow has not run on this fork PR — there's no CI signal for the suite yet.
2. The `FhirPersistanceService` mock injected via ReflectionTestUtils into the singleton service is never restored, so the mock leaks into every later test class in the shared Spring context. An `@After` that restores the original bean is required with this pattern.
3. The save/update tests rely on the real accession generator, which initializes from a static `SpringContext` lookup in `AccessionNumberUtil` — this exact order-dependent NPE was hit on #3852's first iteration; mocking `IAccessionNumberGenerator` (as #3852 now does) avoids it.
4. House style: no wildcard static imports, and the `z_` name prefix is redundant on `@Ignore`d tests.
